services/httpd/config.go

```diff
@@ -28,6 +28,7 @@ type Config struct {
 	WriteTracing          bool          `toml:"write-tracing"`
 	PprofEnabled          bool          `toml:"pprof-enabled"`
 	HTTPSEnabled          bool          `toml:"https-enabled"`
+	HTTPSCACertificate string `toml:"https-ca-certificate"`
```

Comment:
You need to use `go fmt` to format the files. I also don't think this will work for configuration because it needs to be made abundantly clear this is an alternative for authentication. It may be interpreted as a normal client certificate (which should also be supported) to validate a client, but not as authentication and it may get interpreted that way by a casual reading of the options.

So there will need to be one configuration for validating a client certificate to begin with and one that enables reading the CN of the client certificate to perform authentication.

I'm not sure what that looks like right now so just format the file for now.

Reply:
I agree - there should be separate config settings for:

- the certificate to use for verification
- whether client verification is disabled, on but optional, or on and required
- maybe whether to use CN or some other attribute for the username? I didn't find any quick answers when I searched for client certificate usernames.